lib/ain-evm/src/evm.rs

```diff
@@ -252,27 +255,27 @@ impl EVMServices {
     /// across all usages. Note: To be replaced with a proper lock flow later.
     ///
     pub unsafe fn commit_block(&self, template: &BlockTemplate) -> Result<()> {
-        {
-            let Some(BlockData { block, receipts }) = template.block_data.clone() else {
-                return Err(format_err!("no constructed EVM block exist in template id").into());
-            };
-
-            debug!(
-                "[finalize_block] Finalizing block number {:#x}, state_root {:#x}",
-                block.header.number, block.header.state_root
-            );
+        let Some(BlockData { block, receipts }) = template.block_data.clone() else {
+            return Err(format_err!("no constructed EVM block exist in template id").into());
+        };
 
-            self.block.connect_block(&block)?;
-            self.logs
-                .generate_logs_from_receipts(&receipts, block.header.number)?;
-            self.receipt.put_receipts(receipts)?;
-            self.channel
-                .sender
-                .send(Notification::Block(block.header.hash()))
-                .map_err(|e| format_err!(e.to_string()))?;
-        }
-        self.core.clear_account_nonce();
+        debug!(
+            "[finalize_block] Finalizing block number {:#x}, state_root {:#x}",
+            block.header.number, block.header.state_root
+        );
 
+        self.block.connect_block(&block)?;
+        self.logs
+            .generate_logs_from_receipts(&receipts, block.header.number)?;
+        self.receipt.put_receipts(receipts)?;
+        self.channel
+            .sender
+            .send(Notification::Block(block.header.hash()))
+            .map_err(|e| format_err!(e.to_string()))?;
+
+        self.core.clear_account_nonce();
+        self.core.flush()?;
+        self.storage.flush()?;
```

Comment:
Any idea of the impact on sync ? We could do it periodically instead of every block, especially since we add a consistency check with the DVM db

Reply:
Didnt check the perf tradeoff on the sync for this, but we likely may need to do this since we implement a flush on the DVM side on every state update from connect block. If flush happens periodically, unpredictable node crashes will cause state mismatches between the DVM and EVM side, which i guess this PR tradeoffs the consistency for performance

Reply:
Not sure what are your thoughts on this tradeoff?

Reply:
Although on init, we do check for state mismatch between the 2 dbs, but the current pipeline goes into reindex which wont be ideal at all.

Reply:
Sounds good/safe to me.